[GT-169] loader crashes on empty message

Sae126V opened the pull request:
Added a check to raise an error if content is empty when running loader.

"Resolves #304"

Sae126V committed:
[GT-169] Add condition to check content is empty

tofu-rocketry commented on test/test_loader.py:
```diff
@@ -180,9 +180,44 @@ def test_load_all_other_type(self):
 
 if __name__ == '__main__':
-    unittest.main()
+    unittest.main()
```
Can you make sure files follow the Unix convention of having a blank line at the end?

Sae126V replied:
I haven't made this particular change as part of the commit. But, I am happy to fix it.

tofu-rocketry commented on apel/db/loader/loader.py:
```diff
@@ -146,6 +146,17 @@ def load_all_msgs(self):
                 log.info("Loading message %s. ID = %s", self.current_msg, msg_id)
+
+                if (
+                    not data or
```
Similar issue with indentation to this: https://github.com/apel/ssm/pull/228#discussion_r1231209359

Could we go with something like this (from [PEP 8](https://peps.python.org/pep-0008/)) with the line breaks before the logical operator:

```python
# Add some extra indentation on the conditional continuation line.
if (this_is_one_thing
        and that_is_another_thing):
    do_something()
```

Sae126V replied:
Done. Updated the codebase

Sae126V committed:
[GT-169] Update code to fix indentation

tofu-rocketry committed:
Fix whitespace and function name in tests

assertEquals is an old alias that's deprecated in newer Python.

tofu-rocketry committed:
Remove empty 'empaid' check as it's optional

The 'empaid' element of a message is optional, so we shouldn't check if
it's empty otherwise legitimate messages may be rejected. 'body' and
'signer' are both mandatory message elements (once the message has
passed the signer checks in SSM).

tofu-rocketry committed:
Tidy up test cases and improve error message

- Improve the error message when an empty message or message element is
  encountered.
- Add a few more different test cases and group them by good/bad case.

garaimanoj reviewed (approved):
LGTM
